File: lua/triple.lua
for c = 1, 40 do
  for b = 1, c-1 do
    for a = 1, b-1 do
      if a * a + b * b == c * c then
        print(string.format("%d, %d, %d", a, b, c))
      end
    end
  end
end

File: erlang/triple.erl
% First line is ignored when running with escript
main(_) ->
  lists:foreach(fun (C) ->
    lists:foreach(fun (B) ->
      lists:foreach(fun (A) ->
        if
          A*A + B*B == C*C -> io:format("~p, ~p, ~p~n", [A,B,C]);
          true -> ok
        end
      end, lists:seq(1,B))
    end, lists:seq(1,C))
  end, lists:seq(1,40)).
